java/vector/src/test/java/org/apache/arrow/vector/file/json/TestJSONFile.java

```diff
@@ -96,28 +97,25 @@ public void testWriteReadUnionJSON() throws IOException {
     try (
         BufferAllocator vectorAllocator = allocator.newChildAllocator("original vectors", 0, Integer.MAX_VALUE);
         NullableMapVector parent = NullableMapVector.empty("parent", vectorAllocator)) {
-
       writeUnionData(count, parent);
-
       printVectors(parent.getChildrenFromFields());
 
-      VectorSchemaRoot root = new VectorSchemaRoot(parent.getChild("root"));
-      validateUnionData(count, root);
+      try (VectorSchemaRoot root = new VectorSchemaRoot(parent.getChild("root"))) {
```

Comment:
why changing to a nested try blocks?

Reply:
Because I need both `VectorSchemaRoot` to compare equality.